src/test/java/org/zalando/pazuzu/ApiAuthTest.java

```diff
@@ -0,0 +1,282 @@
+package org.zalando.pazuzu;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.web.FilterChainProxy;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.zalando.pazuzu.mock.ResourceServerTokenServicesMock;
+import org.zalando.stups.oauth2.spring.server.DefaultAuthenticationExtractor;
+import org.zalando.stups.oauth2.spring.server.TokenInfoResourceServerTokenServices;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = { PazuzuAppLauncher.class, ResourceServerTokenServicesMock.class })
+@WebIntegrationTest(randomPort = true)
+public class ApiAuthTest extends AbstractComponentTest {
+    @Autowired
+    private TokenInfoResourceServerTokenServices resourceServerTokenServices;
+    @Autowired
+    private WebApplicationContext context;
+    @Autowired
+    private FilterChainProxy springSecurityFilterChain;
+
+    private MockMvc mockMvc;
+
+    private static final String user = "user";
+    private static final String admin = "test";
+    private static final String VALID_TOKEN = "TOKEN_1234567890";
+
+    private static final String body = "{\n" +
+            "  \"name\": \"python\",\n" +
+            "  \"docker_data\": \"RUN python\",\n" +
+            "  \"test_instruction\": \"python -V\",\n" +
+            "  \"dependencies\": []\n" +
+            "}";
+
+    private ObjectMapper mapper;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'mapper' could be replaced by a local variable.](https://www.codacy.com/app/Pazuzu/pazuzu-registry/file/2725695128/issues/source?bid=3131965&fileBranchId=3348590#l61)